service/history/queues/slice.go

```diff
@@ -361,10 +407,31 @@ func (s *SliceImpl) SelectTasks(batchSize int) ([]Executable, error) {
 
 func (s *SliceImpl) destroy() {
 	s.destroyed = true
+	s.iterators = nil
+	s.outstandingExecutables = nil
 }
 
-func (s *SliceImpl) validateNotDestroyed() {
+func (s *SliceImpl) stateSanityCheck() {
 	if s.destroyed {
 		panic("Can not invoke method on destroyed queue slice")
 	}
 }
+
+func validateIteratorsOrderedDisjoint(
+	iterators []Iterator,
+) {
+	if len(iterators) <= 1 {
+		return
+	}
+
+	for idx, iterator := range iterators[:len(iterators)-1] {
+		nextIterator := iterators[idx+1]
+		if iterator.Range().ExclusiveMax.CompareTo(nextIterator.Range().InclusiveMin) >= 0 {
```

Comment:
should this be `> 0` instead of `>= 0`?

Reply:
Since iterator doesn't have predicate, two iterators should be merged together if one's max equals the other's min. As iterator.CanMerge() returns true in the case, I think the check should panic when we found they're not merged together. I can update the panic message to be more specific.